Zyantific.Zydis/Native/DecoderTypes.cs

```diff
@@ -23,6 +40,457 @@ public enum AddressWidth
+        EXCEPTION_CLASS_AVX2,
+        EXCEPTION_CLASS_AVX3,
+        EXCEPTION_CLASS_AVX4,
+        EXCEPTION_CLASS_AVX5,
+        EXCEPTION_CLASS_AVX6,
+        EXCEPTION_CLASS_AVX7,
+        EXCEPTION_CLASS_AVX8,
+        EXCEPTION_CLASS_AVX11,
+        EXCEPTION_CLASS_AVX12,
+        EXCEPTION_CLASS_E1,
+        EXCEPTION_CLASS_E1NF,
+        EXCEPTION_CLASS_E2,
+        EXCEPTION_CLASS_E2NF,
+        EXCEPTION_CLASS_E3,
+        EXCEPTION_CLASS_E3NF,
+        EXCEPTION_CLASS_E4,
+        EXCEPTION_CLASS_E4NF,
+        EXCEPTION_CLASS_E5,
+        EXCEPTION_CLASS_E5NF,
+        EXCEPTION_CLASS_E6,
+        EXCEPTION_CLASS_E6NF,
+        EXCEPTION_CLASS_E7NM,
+        EXCEPTION_CLASS_E7NM128,
+        EXCEPTION_CLASS_E9NF,
+        EXCEPTION_CLASS_E10,
+        EXCEPTION_CLASS_E10NF,
+        EXCEPTION_CLASS_E11,
+        EXCEPTION_CLASS_E11NF,
+        EXCEPTION_CLASS_E12,
+        EXCEPTION_CLASS_E12NP,
+        EXCEPTION_CLASS_K20,
+        EXCEPTION_CLASS_K21,
+        EXCEPTION_CLASS_MAX_VALUE = EXCEPTION_CLASS_K21,
+    }
+
+    public enum MaskMode
+    {
+        MASK_MODE_INVALID,
+        MASK_MODE_DISABLED,
+        MASK_MODE_MERGING,
+        MASK_MODE_ZEROING,
+        MASK_MODE_CONTROL,
+        MASK_MODE_CONTROL_ZEROING,
+        MASK_MODE_MAX_VALUE = MASK_MODE_CONTROL_ZEROING,
+    }
+
+    public enum BroadcastMode
+    {
+        BROADCAST_MODE_INVALID,
+        BROADCAST_MODE_1_TO_2,
+        BROADCAST_MODE_1_TO_4,
+        BROADCAST_MODE_1_TO_8,
+        BROADCAST_MODE_1_TO_16,
+        BROADCAST_MODE_1_TO_32,
+        BROADCAST_MODE_1_TO_64,
+        BROADCAST_MODE_2_TO_4,
+        BROADCAST_MODE_2_TO_8,
+        BROADCAST_MODE_2_TO_16,
+        BROADCAST_MODE_4_TO_8,
+        BROADCAST_MODE_4_TO_16,
+        BROADCAST_MODE_8_TO_16,
+        BROADCAST_MODE_MAX_VALUE = BROADCAST_MODE_8_TO_16,
+    }
+
+    public enum RoundingMode
+    {
+        ROUNDING_MODE_INVALID,
+        ROUNDING_MODE_RN,
+        ROUNDING_MODE_RD,
+        ROUNDING_MODE_RU,
+        ROUNDING_MODE_RZ,
+        ROUNDING_MODE_MAX_VALUE = ROUNDING_MODE_RZ,
+    }
+
+    public enum SwizzleMode
+    {
+        SWIZZLE_MODE_INVALID,
+        SWIZZLE_MODE_DCBA,
+        SWIZZLE_MODE_CDAB,
+        SWIZZLE_MODE_BADC,
+        SWIZZLE_MODE_DACB,
+        SWIZZLE_MODE_AAAA,
+        SWIZZLE_MODE_BBBB,
+        SWIZZLE_MODE_CCCC,
+        SWIZZLE_MODE_DDDD,
+        SWIZZLE_MODE_MAX_VALUE = SWIZZLE_MODE_DDDD,
+    }
+
+    public enum ConversionMode
+    {
+        CONVERSION_MODE_INVALID,
+        CONVERSION_MODE_FLOAT16,
+        CONVERSION_MODE_SINT8,
+        CONVERSION_MODE_UINT8,
+        CONVERSION_MODE_SINT16,
+        CONVERSION_MODE_UINT16,
+        CONVERSION_MODE_MAX_VALUE = CONVERSION_MODE_UINT16,
+    }
+
+    public enum PrefixType
+    {
+        PREFIX_TYPE_IGNORED,
+        PREFIX_TYPE_EFFECTIVE,
+        PREFIX_TYPE_MANDATORY,
+        PREFIX_TYPE_MAX_VALUE = PREFIX_TYPE_MANDATORY,
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
```

Comment:
I would probably like to nest the struct definitions to mirror the `C` layout.

```c#
public struct DecodedInstruction
{
    public struct Raw
    {
        ...
    }

    public ZyanU8 length;

    ...

    public Raw Raw;
}
```

This way we could shorten some names like `DecodedInstructionRawDisp`.